packages/core/src/routes/sign-in-experience.ts

```diff
@@ -16,7 +17,16 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
    */
   router.get('/sign-in-exp', async (ctx, next) => {
     const signInExperience = await findDefaultSignInExperience();
-    ctx.body = signInExperience;
+    const { socialSignInConnectorIds: selectedSocialSignInConnectorIds } = signInExperience;
+    const enabledSocialConnectorIds = new Set(await getEnabledSocialConnectorIds());
```

Comment:
why do we net a set here?

Reply:
`set.has()` is better than `list.has` in performance.

```ts
    const socialSignInConnectorIds = selectedSocialSignInConnectorIds.filter((id) =>
      enabledSocialConnectorIds.has(id)
    );
```